pkg/collector/py/kubeutil.go

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build cpython,kubelet
+
+package py
+
+import (
+	"time"
+	"unsafe"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+)
+
+// #cgo pkg-config: python-2.7
+// #cgo linux CFLAGS: -std=gnu99
+// #include "api.h"
+// #include "kubeutil.h"
+import "C"
+
+var kubeletCacheKey = cache.BuildAgentKey("py", "kubeutil", "connection_info")
+
+// GetKubeletConnectionInfo returns a dict containing url and credentials to connect to the kubelet.
+// The dict is empty if the kubelet was not detected. The call to kubeutil is cached for 5 minutes.
+// See the documentation of kubelet.GetRawConnectionInfo for dict contents.
+//export GetKubeletConnectionInfo
+func GetKubeletConnectionInfo() *C.PyObject {
+	var creds map[string]string
+	var ok bool
+	dict := C.PyDict_New()
+
+	if cached, hit := cache.Cache.Get(kubeletCacheKey); hit {
+		creds, ok = cached.(map[string]string)
+		if !ok {
+			log.Errorf("invalid cache format, forcing a cache miss")
+			creds = nil
+		}
+	}
+
+	if creds == nil { // cache miss
+		kubeutil, err := kubelet.GetKubeUtil()
+		if err != nil {
+			return dict
+		}
+		creds = kubeutil.GetRawConnectionInfo()
```

Comment:
what happens if the kubelet is not found (because it hasn't started yet for example)? Can we cache for less than 5min in this case? Ideally we'd only cache for like 10 sec until we permanently fail the kubeutil init. Otherwise an initial failure (due to the agent starting before the kubelet, or a network blip) means the check will be disabled for 5 minutes.

Reply:
If the kubelet is not found, `kubelet.GetKubeUtil()` will return an error, and we will return an empty dict without setting it in cache, so we'll retry on the next check run. Its lifecycle is different than its agent5's cousin, I'll add a comment to make it clear